examples/aggregator/module/rust/src/lib.rs

```diff
@@ -35,221 +36,49 @@ pub mod proto {
-                        Err(err) => {
-                            let err = format!("Submit sample error: {}", err);
-                            Err(grpc::build_status(grpc::Code::InvalidArgument, &err))
-                        }
-                    }
-                }
-                Err(err) => {
-                    let err = format!("Data deserialization error: {}", err);
-                    Err(grpc::build_status(grpc::Code::InvalidArgument, &err))
-                }
-            },
-            None => {
-                let err = "No data specified";
-                Err(grpc::build_status(grpc::Code::InvalidArgument, &err))
-            }
-        }
-    }
-}
-
-oak::entrypoint!(aggregator<Either<AggregatorInit, oak::grpc::Invocation>> => |receiver: Receiver<Either<AggregatorInit, oak::grpc::Invocation>>| {
-    oak::logger::init_default();
-
-    let mut receiver_iterator = receiver.iter();
-
-    // Receive the initialization message.
-    let init_message: AggregatorInit = receiver_iterator
-        .find_map(Either::left)
-        .expect("Couldn't receive init message");
-    let grpc_client_invocation_sender = init_message
-        .grpc_client_invocation_sender
-        .expect("Couldn't receive gRPC invocation sender")
-        .sender
-        .expect("Empty gRPC invocation sender");
-
-    // Run event loop and handle incoming invocations.
-    let node = AggregatorNode::new(grpc_client_invocation_sender);
-    let dispatcher = AggregatorDispatcher::new(node);
-    oak::run_command_loop(dispatcher, receiver_iterator.filter_map(Either::right));
-});
 
 #[derive(Debug, serde::Deserialize)]
 #[serde(deny_unknown_fields)]
 pub struct Config {
     pub grpc_server_listen_address: String,
+    pub backend_server_address: String,
+    pub aggregator_module_hash: String,
 }
 
-/// Create initialization message for Aggregator Node.
-fn create_init_message(sender: oak::io::Sender<GrpcInvocation>) -> AggregatorInit {
-    AggregatorInit {
-        grpc_client_invocation_sender: Some(GrpcInvocationSender {
-            sender: Some(sender),
-        }),
+/// Main entrypoint of the Aggregator application.
+///
+/// This node is in charge of creating the other top-level nodes, but does not process any request.
+#[derive(Default)]
+struct Main;
+
+impl oak::CommandHandler for Main {
+    type Command = ConfigMap;
+
+    fn handle_command(&mut self, command: ConfigMap) -> anyhow::Result<()> {
+        let config: Config =
+            toml::from_slice(&command.items.get("config").expect("Couldn't find config"))
+                .context("Couldn't parse TOML config file")?;
+        info!("Parsed config: {:?}", config);
+
+        let router_init_sender =
+            oak::io::entrypoint_node_create::<Router>("router", &Label::public_untrusted(), "app")
+                .context("Couldn't create router node")?;
+        let router_command_sender =
+            oak::io::send_init(router_init_sender, command, &Label::public_untrusted())?;
```

Comment:
Am I reading this correctly that it is sending the entire configmap to the router node? That seems excessive, also the config in general may contain other details that are irrelevant for the router node. AFAICT the only reason for passing it is in order to create the gRPC client node (which I think should be created here) and in order to pass the wasm module hash (which I think should be hardcoded here, though that's not possible yet because the app is still in a single wasm module for now). Maybe leave a TODO for the latter for now? Though I think the former could be fixed now already.

Reply:
Adding a TODO(https://github.com/project-oak/oak/issues/1731)